### ISSUE TITLE
Cannot get SSL to work with both www and non www domain

### ISSUE THREAD

**Issue by baldlion** [TryGhost/Ghost-CLI#394]
* [ ] Bug Report
* [x] Feature Request

### Summary

I need to configure Nginx to direct all www and http requests to https://domain.com. I have not been able to get this working, even after following several articles on the subject (non-ghost specific).

The following urls work:

- example.com
- https://example.com

The following urls do not work (Chrome is blocking my connection with a security error):

- www.example.com
- https://www.example.com

![image](https://user-images.githubusercontent.com/22313496/28736258-0255713c-73b7-11e7-88a7-5c559fcc268e.png)

I have attempted to get this working by adding these server blocks in my domain.com-ssl.conf file:

```
server {
    listen 80;
    server_name www.example.com;
    return 301 https://example.com$request_uri;
}

server {
    listen 443;
    server_name www.example.com;

    ssl on;
    ssl_certificate /home/user/.acme.sh/example.com/fullchain.cer;
    ssl_certificate_key /home/user/.acme.sh/example.com/example.com.key;
    include /var/www/example.com/system/files/ssl-params.conf;


    return 301 https://example.com$request_uri;
}

server {
    listen 80;
    server_name example.com;
    return 301 https://example.com$request_uri;
}
```

* OS: Ubuntu 16.04.2 (Digital Ocean)
* Node Version: 6.11.1
* Ghost-CLI Version: 1.0.0
* Environment:  Production

**Comment by acburdine** [comment]
This error is not surprising, as an ssl cert for example.com works for ONLY example.com, www.example.com is a completely separate domain for the purposes of SSL, and thus the cert doesn't work.

Easiest way to do this at present is to manually create another ssl cert using [acme.sh](https://github.com/NeilPang/acme.sh) - and then manually update your nginx config to use the www ssl certs for the redirect.

Going to close this though as it's in most respects a duplicate of #224

**Comment by baldlion** [comment]
@acburdine I am getting an invalid response from acme.sh when trying to create a new certificate for www.example.com (because the existing certificate created through ghost-cli is blocking the request to www?). Do you know a workaround?

**Comment by acburdine** [comment]
You've modified your nginx configs quite a bit - what you'll have to do is modify the port 80 config for the www domain to look something like this:

```nginx
server {
   listen 80;
   server_name www.example.com;
   
   # this should match the root config in your ssl config for example.com  
   root /path/to/ghost/install/system/nginx-root; 
   
   location ~ /.well-known {
        allow all;
   }
}
```

**Comment by baldlion** [comment]
I am still getting an invalid response error from acme.sh. I cleaned up my example.com-ssl.conf to look like this:

```
server {
    listen 80;
    server_name www.example.com;

    root /var/www/example.com/system/nginx-root;

    location ~ /.well-known {
        allow all;
    }
}

server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;

    server_name example.com;
    root /var/www/example.com/system/nginx-root;

    ssl_certificate /home/user/.acme.sh/example.com/fullchain.cer;
    ssl_certificate_key /home/user/.acme.sh/example.com/example.com.key;
    include /var/www/example.com/system/files/ssl-params.conf;

    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $http_host;
        proxy_pass http://127.0.0.1:2368;

    }

    location ~ /.well-known {
        allow all;
    }

    client_max_body_size 50m;
}
```

**Comment by bahagiga** [comment]
@baldlion [this](https://gist.github.com/robertnealan/891fe8cf3b7541e922e3b39e9ae9fd96#file-sites-available-ghost) worked for me pre-ghost 1.0

**Comment by baldlion** [comment]
@bahagiga thank you! I was able to get this working following the steps [here](https://robertnealan.com/setting-up-ssl-for-ghost-on-digitalocean-with-lets-encrypt/). Instead of steps 6 and 7, I used acme.sh (steps 2 and 3 [here](https://github.com/Neilpang/acme.sh)).
